monitor/routerecon/routescan.go

```diff
@@ -54,7 +58,7 @@ type filter interface {
 	Match(tx crossTxJSON) (bool, error)
 }
 
-func queryLatestCrossTx(ctx context.Context, network netconf.ID, filter filter, next string) (crossTxJSON, string, error) {
+func queryLatestCrossTx(ctx context.Context, network netconf.ID, routeScanAPIKey string, filter filter, next string) (crossTxJSON, string, bool, error) {
```

Comment:
not a fan of changing the function signature jsut for testing purposes. 

What about:
```
// responseHook allows internal tests to verify raw http responses.
// The default response hook is a noop.
var responseHook = func(*http.Response){}
```
then set this variable in tests accordingly.

Reply:
also avoid stuttering